src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session.cs

```diff
@@ -64,12 +64,35 @@ internal void UpdateModelTrackingSpan(SnapshotPoint initialPosition)
                 });
             }
 
-            public override void Stop()
+            /// <summary>
+            /// Dismisses the editor completion list.
+            /// 
+            /// NOTE: If calling this, do not call <see cref="CommitEditorSession"/>.
+            /// </summary>
+            public override void DismissEditorSession()
             {
                 AssertIsForeground();
+                this.DisconnectFromPresenter();
+                base.DismissEditorSession();
+            }
+
+            /// <summary>
+            /// Commits and dismisses the underlying editor completion list.
+            /// 
+            /// NOTE: If calling this, do not call <see cref="DismissEditorSession"/>.
+            /// Committing the editor list will implicitly dismiss it.
+            /// </summary>
+            public void CommitEditorSession()
+            {
+                AssertIsForeground();
+                this.DisconnectFromPresenter();
+                this.PresenterSession.Commit();
+            }
+
+            private void DisconnectFromPresenter()
+            {
                 this.PresenterSession.ItemSelected -= OnPresenterSessionItemSelected;
                 this.PresenterSession.ItemCommitted -= OnPresenterSessionItemCommitted;
-                base.Stop();
```

Comment:
Is it ok that we're not calling into the base here?  i.e. DismissEditorSession calls Base.Dismiss...  i assume that will cause 'base.Stop' to be called.  But that's not the case for CommitEditorSession?  Is it bad that we're missing a call to base.Stop here?